lib/src/shared/main/cpp/native_c4.cc

```diff
@@ -27,6 +33,37 @@ using namespace litecore::jni;
 static jclass cls_C4Log;
 static jmethodID m_C4Log_logCallback;
 
+// The default logging callback writes to stderr, or on Android to __android_log_write.
+void vPlatformLog(int level, const char *fmt, va_list args) {
+#if defined(__ANDROID__)
+    static const int androidLevels[5]
+            = {ANDROID_LOG_VERBOSE, ANDROID_LOG_DEBUG, ANDROID_LOG_INFO, ANDROID_LOG_WARN, ANDROID_LOG_ERROR};
+    __android_log_vprint(androidLevels[level], "LiteCore/JNI", fmt, args);
+#else
+    static const char* const kLevels[] = {"V", "D", "I", "W", "E",};
+
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+
+    struct tm tm;
+    localtime_r(&tv.tv_sec, &tm);
+
+    char timestamp[100];
+    strftime(timestamp, sizeof(timestamp), "%T", &tm);
+
+    fprintf(stderr, "%s.%03u %s/LiteCore/JNI: ", timestamp, tv.tv_usec / 1000, kLevels[level]);
+    vfprintf(stderr, fmt, args);
```

Comment:
If there is anything special to be done for a web server environment, perhaps do it here.

Reply:
I don't know.  Added comment to that effect.